src/axom/primal/operators/detail/intersect_bounding_box_impl.hpp

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef PRIMAL_INTERSECT_BOUNDING_BOX_HPP_
+#define PRIMAL_INTERSECT_BOUNDING_BOX_HPP_
+
+#include "axom/core/Macros.hpp"
+
+#include <type_traits> // for std::is_floating_point< T >()
+
+namespace axom
+{
+namespace primal
+{
+namespace detail
+{
+
+/*!
+ * \brief Helper routine for AABB / AABB intersection test.
+ *
+ * \param [in] min1 the 1st AABB min coordinate along a direction.
+ * \param [in] max1 the 1st AABB max coordinate along a drection.
+ * \param [in] min2 the 2nd AABB min coordinate along a direction.
+ * \param [in] max2 the 2nd AABB max coordinate along a drection.
+ *
+ * \return status true if the AABB's intersect, otherwise false.
+ *
+ * \note This routine is called by the intersect AABB/AABB methods for each
```

Comment:
If there is no fuzzy intersection, It might be worth mentioning that that the AABBs can be scaled which essentially creates a fuzzy intersetion.

Reply:
Good point @mclarsen -- Since this method is in primal and can be used outside of the BVH, I think it may make sense for this method to take an optional tolerance

Reply:
@kennyweiss , @mclarsen  - Good to know! I added a doxygen comment to use scaling for fuzzy intersections.